app/Domain/Validation/Rules/DoesNotExceedLimitRule.php

```diff
@@ -41,13 +41,16 @@ public function errorMessage(): string
 
     protected function getUserVacationLimit(User $user, YearPeriod $yearPeriod): int
     {
-        return $user->vacationLimits()->where("year_period_id", $yearPeriod->id)->first()->days ?? 0;
+        return $user->vacationLimits()
+            ->whereBelongsTo($yearPeriod)
+            ->first()
+            ->days ?? 0;
```

Comment:
I think `?` is missing
```?->days ?? 0```
Cause `first()` method may return null, then you will receive something like "call property on null"